### PR TITLE
[FL-3055] Getter for application data path

### PR DESCRIPTION
**Pull request by DrZlo13**
# What's new

- Storage: apps_data path getter, that ensures the path exists.

# Verification 

- Launch spi_mem_manager, music_player, picopass, hid_ble.
- Make sure the correct folder is created in /ext/apps_data and the app is using that folder.
- Make sure the data from the old application path is copied to the new path and the old path is deleted.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/debug/unit_tests/storage/storage_test.c [hunk 1]
```diff
@@ -303,9 +342,256 @@ MU_TEST_SUITE(storage_rename) {
...
 
+#define APPSDATA_APP_PATH(path) APPSDATA_PATH "/" path
+
+static const char* storage_test_apps[] = {
+    "-_twilight_-",
+    "-_rainbow_-",
+    "-_pinkie_-",
+    "-_apple_-",
+    "-_flutter_-",
+    "-_rare_-",
```
Comment (hedger) [inline applications/debug/unit_tests/storage/storage_test.c:357]: ICWYDT

File: applications/examples/example_apps_data/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Apps Data folder Example
+
+This example demonstrates how to utilize the Apps Data folder to store data that is not part of the app itself, such as user data, configuration files, and so forth.
+
+## What is the Apps Data Folder?
+
+The **Apps Data** folder is a folder used to store data for external apps that are not part of the main firmware. 
+
+The path to the current application folder is related to the `appsid` of the app. The `appsid` is used to identify the app in the app store and is stored in the `application.fam` file.
```
Comment (hedger) [inline applications/examples/example_apps_data/README.md]: appid

File: applications/examples/example_apps_data/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+This example demonstrates how to utilize the Apps Data folder to store data that is not part of the app itself, such as user data, configuration files, and so forth.
+
+## What is the Apps Data Folder?
+
+The **Apps Data** folder is a folder used to store data for external apps that are not part of the main firmware. 
+
+The path to the current application folder is related to the `appsid` of the app. The `appsid` is used to identify the app in the app store and is stored in the `application.fam` file. 
+The Apps Data folder is located only on the external storage, the SD card.
+
+For example, if the `appsid` of the app is `snake_game`, the path to the Apps Data folder will be `/ext/apps_data/snake_game`. But using raw paths is not recommended, because the path to the Apps Data folder can change in the future. Use the `/app` alias instead.
```
Comment (hedger) [inline applications/examples/example_apps_data/README.md]: appid

File: applications/examples/example_apps_data/example_apps_data.c [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    File* file = storage_file_alloc(storage);
+
+    // Get the path to the current application data folder
+    // That is: /ext/apps_data/<app_name>
+    // And it will create folders in the path if they don't exist
+    // In this example it will create /ext/apps_data/example_apps_data
+    // And file will be /ext/apps_data/example_apps_data/test.txt
+
+    // Open file, write data and close it
+    if(!storage_file_open(file, "/app/test.txt", FSAM_WRITE, FSOM_CREATE_ALWAYS)) {
```
Comment (hedger) [inline applications/examples/example_apps_data/example_apps_data.c]: APPS_DATA_PATH()

File: applications/plugins/picopass/picopass.c [hunk 5]
```diff
@@ -171,6 +171,12 @@ void picopass_show_loading_popup(void* context, bool show) {
     }
 }
 
+static void picopass_migrate_from_old_folder() {
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    storage_common_migrate(storage, "/ext/picopass", STORAGE_APPS_DATA_PATH_PREFIX);
```
Comment (hedger) [inline applications/plugins/picopass/picopass.c]: EXT_PATH("picopass")

File: applications/plugins/spi_mem_manager/spi_mem_app.c [hunk 6]
```diff
@@ -37,7 +37,8 @@ SPIMemApp* spi_mem_alloc(void) {
     instance->text_input = text_input_alloc();
     instance->mode = SPIMemModeUnknown;
 
-    furi_string_set(instance->file_path, SPI_MEM_FILE_FOLDER);
+    // Migrate data from old sd-card folder
+    storage_common_migrate(instance->storage, "/ext/spimem", STORAGE_APPS_DATA_PATH_PREFIX);
```
Comment (hedger) [inline applications/plugins/spi_mem_manager/spi_mem_app.c]: EXT_PATH("spimem")

File: applications/services/bt/bt_service/bt_api.c [hunk 7]
```diff
@@ -45,7 +45,14 @@ void bt_keys_storage_set_storage_path(Bt* bt, const char* keys_storage_path) {
     furi_assert(bt->keys_storage);
     furi_assert(keys_storage_path);
 
-    bt_keys_storage_set_file_path(bt->keys_storage, keys_storage_path);
+    Storage* storage = furi_record_open("storage");
```
Comment (hedger) [inline applications/services/bt/bt_service/bt_api.c]: RECORD_STORAGE

File: applications/services/bt/bt_service/bt_api.c [hunk 8]
```diff
@@ -45,7 +45,14 @@ void bt_keys_storage_set_storage_path(Bt* bt, const char* keys_storage_path) {
...
+    FuriString* path = furi_string_alloc_set(keys_storage_path);
+    storage_common_process_aliases(storage, path);
+
+    bt_keys_storage_set_file_path(bt->keys_storage, furi_string_get_cstr(path));
+
+    furi_string_free(path);
+    furi_record_close("storage");
```
Comment (hedger) [inline applications/services/bt/bt_service/bt_api.c]: RECORD_STORAGE

File: applications/services/dialogs/dialogs_api.c [hunk 9]
```diff
@@ -13,6 +14,22 @@ bool dialog_file_browser_show(
     FuriApiLock lock = api_lock_alloc_locked();
     furi_check(lock != NULL);
 
+    Storage* storage = furi_record_open("storage");
```
Comment (hedger) [inline applications/services/dialogs/dialogs_api.c]: RECORD_STORAGE

File: applications/services/dialogs/dialogs_api.c [hunk 10]
```diff
@@ -39,6 +56,9 @@ bool dialog_file_browser_show(
         furi_message_queue_put(context->message_queue, &message, FuriWaitForever) == FuriStatusOk);
     api_lock_wait_unlock_and_free(lock);
 
+    furi_record_close("storage");
```
Comment (hedger) [inline applications/services/dialogs/dialogs_api.c]: RECORD_STORAGE

**Review by hedger — CHANGES_REQUESTED** [review]

**Review by hedger — DISMISSED** [review]

**Review by hedger — APPROVED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**